# Tópico 1 — Localização e Download do Dataset Público

### Dataset: **European Soccer Database**

🔗 **Link:** [https://www.kaggle.com/datasets/hugomathien/soccer](https://www.kaggle.com/datasets/hugomathien/soccer)

---

## 📄 Sobre o Dataset

O *European Soccer Database* é um dos maiores conjuntos de dados públicos sobre futebol na Europa, contendo informações completas para análises e aplicações de machine learning.

### 📊 O que está incluído:

* **+25.000** partidas
* **+10.000** jogadores
* **11 países europeus** com suas principais ligas
* Temporadas de **2008 a 2016**
* Atributos de jogadores e times (originados do FIFA e atualizados semanalmente)
* Formações e coordenadas (X,Y) dos jogadores em campo
* Odds de apostas de até **10 casas diferentes**
* Eventos detalhados de partidas

  * Tipos de gol
  * Posse de bola
  * Escanteios
  * Cruzamentos
  * Faltas
  * Cartões
  * E muito mais
* *(Atualização de 16/10/2016: nova tabela contendo atributos completos dos times do FIFA)*


## Tópico 2 - Importação dos Dados em Python


In [6]:
import sqlite3
import pandas as pd

##Caminho para o arquivo SQLite
db_path = "../DataBase/database.sqlite"

##Conectar ao banco
conn = sqlite3.connect(db_path)

##Listar todas as tabelas
tables = pd.read_sql("SELECT name FROM sqlite_master WHERE type='table';", conn)
tables

name
0    sqlite_sequence
1  Player_Attributes
2             Player
3              Match
4             League
5            Country
6               Team
7    Team_Attributes

In [8]:
##Ler uma tabela específica (exemplo: Times da Premier League)
df_queryResult = pd.read_sql("SELECT DISTINCT team_long_name FROM Team WHERE team_api_id IN (SELECT home_team_api_id FROM Match WHERE league_id = (SELECT id FROM League WHERE name = 'England Premier League') UNION SELECT away_team_api_id FROM Match WHERE league_id = (SELECT id FROM League WHERE name = 'England Premier League'));", conn)

##Mostrar primeiras linhas
df_queryResult.head(50)

team_long_name
0                   Burnley
1            Leicester City
2              Cardiff City
3                   Chelsea
4           Manchester City
5                Portsmouth
6               Southampton
7                Sunderland
8                 Blackpool
9            Wigan Athletic
10            Middlesbrough
11         Bolton Wanderers
12        Tottenham Hotspur
13  Wolverhampton Wanderers
14                Liverpool
15          West Ham United
16         Blackburn Rovers
17          Birmingham City
18     West Bromwich Albion
19                Hull City
20                  Everton
21              Bournemouth
22                  Reading
23                  Watford
24                  Arsenal
25           Crystal Palace
26             Norwich City
27                   Fulham
28             Swansea City
29      Queens Park Rangers
30               Stoke City
31              Aston Villa
32        Manchester United
33         Newcastle United

## Tópico 3 - Análise exploratória dos dados (dimensões, tipos, primeiras linhas, estatísticas descritivas).

## 📊 **Estrutura da Análise Exploratório**

### 1. **Dimensões**

```python
df_tabela.shape
```

* **Objetivo**: Mostrar o número de **linhas** e **colunas** da tabela.
* **Significado**:

  * **Linhas**: Quantos registros (ou entradas) estão presentes na tabela.
  * **Colunas**: Quantas variáveis (ou atributos) a tabela possui.

**Exemplo**:

```python
(11060, 8)
```

Aqui, temos 11.060 registros e 8 variáveis (colunas) na tabela.

---

### 2. **Tipos das Colunas**

```python
df_tabela.dtypes
```

* **Objetivo**: Exibir o tipo de dado de cada coluna, como `int64`, `float64`, `object`, etc.
* **Significado**:

  * Isso ajuda a entender **como os dados estão armazenados** (numéricos, texto, datas, etc.).

**Exemplo**:

```python
id                      int64
player_api_id           int64
player_name            object
player_fifa_api_id     int64
birthday               object
height                float64
weight                float64
dtype: object
```

Aqui, vemos que **id, player_api_id, player_fifa_api_id** são numéricos (`int64`), enquanto **player_name, birthday** são **texto** (`object`), e **height, weight** são **números com ponto flutuante** (`float64`).

---

### 3. **Primeiras Linhas**

```python
df_tabela.head(5)
```

* **Objetivo**: Exibir as **primeiras 5 linhas** da tabela para ter uma visão rápida dos dados.
* **Significado**: Isso ajuda a **visualizar** os dados reais e garantir que tudo esteja conforme esperado.

**Exemplo**:

```python
   id  player_api_id player_name     player_fifa_api_id  birthday    height   weight
0   1        2625.0    Danilo              165664.91   1989-03-02     181.87    168.38
1   2        2626.0    Messi               165665.50   1987-07-24     169.80    140.00
2   3        2627.0    Neymar              165666.00   1992-02-05     170.80    130.00
3   4        2628.0    Ronaldo             165667.12   1985-06-23     180.50    150.00
4   5        2629.0    Mbappé              165668.30   1999-12-08     175.00    160.00
```

---

### 4. **Estatísticas Descritivas**

Aqui, vamos **explicar cada linha da tabela de estatísticas descritivas** gerada por:

```python
df_tabela.describe(include="all")
```

A função `describe()` gera várias **estatísticas resumo** para cada coluna numérica e categórica. Vamos detalhar cada linha da saída.

#### Explicação das linhas:

| **Linha**  | **Significado**                                                                                                                                    |
| ---------- | -------------------------------------------------------------------------------------------------------------------------------------------------- |
| **count**  | **Número de valores não nulos** na coluna. Exemplo: se a coluna "height" tiver valores ausentes, o count mostrará quantos valores válidos existem. |
| **unique** | Número de **valores únicos** em uma coluna. Usado principalmente para colunas categóricas (como "player_name").                                    |
| **top**    | O **valor mais frequente** (modo) na coluna. Por exemplo, qual jogador aparece mais vezes no dataset.                                              |
| **freq**   | A **frequência** do valor mais comum na coluna. Se "Danilo" é o nome mais repetido, o `freq` mostrará quantas vezes ele aparece.                   |
| **mean**   | **Média** dos valores numéricos na coluna. Exemplo: a média das alturas dos jogadores.                                                             |
| **std**    | O **desvio padrão** dos valores, mostrando a **dispersão** dos dados. Valores mais altos indicam maior variação.                                   |
| **min**    | O **menor valor** registrado na coluna (para colunas numéricas). Exemplo: a altura mais baixa dos jogadores.                                       |
| **25%**    | O **1º quartil** (25%) dos dados, ou seja, 25% dos valores estão abaixo desse ponto.                                                               |
| **50%**    | O **2º quartil** (50%) ou **mediana**. Metade dos valores estão abaixo e metade acima desse valor.                                                 |
| **75%**    | O **3º quartil** (75%) dos dados, ou seja, 75% dos valores estão abaixo desse ponto.                                                               |
| **max**    | O **maior valor** registrado na coluna (para colunas numéricas). Exemplo: a altura mais alta dos jogadores.                                        |

---

### Exemplo de Saída:

Suponha que a tabela de estatísticas para a coluna `height` seja assim:

```
       height
count   11060.000000
unique         NaN
top            NaN
freq           NaN
mean      181.867445
std          6.369201
min        157.480000
25%        177.800000
50%        182.880000
75%        185.420000
max        208.280000
```

#### Como interpretar:

1. **count**: Há 11.060 jogadores com altura registrada.
2. **unique**: Não se aplica aqui, pois "height" é numérica.
3. **top**: Não se aplica aqui, pois "height" não tem um valor mais frequente.
4. **freq**: Também não se aplica a números.
5. **mean**: A **média da altura** dos jogadores é **181.87 cm**.
6. **std**: O **desvio padrão** é **6.37 cm**, indicando que a altura dos jogadores tem uma variação de aproximadamente 6 cm.
7. **min**: O jogador mais baixo tem **157.48 cm**.
8. **25%**: 25% dos jogadores têm altura **menor ou igual a 177.8 cm**.
9. **50%**: A mediana da altura dos jogadores é **182.88 cm** (meio dos dados).
10. **75%**: 75% dos jogadores têm altura **menor ou igual a 185.42 cm**.
11. **max**: O jogador mais alto tem **208.28 cm**.


In [17]:
# Lista de tabelas que você quer excluir da análise
tabelas_excluidas = ['sqlite_sequence', 'Player_Attributes', 'Team_Attributes']

tabelas_filtradas = tables[~tables['name'].isin(tabelas_excluidas)]

for tabela in tabelas_filtradas['name']:
    print(f"Analisando a tabela: {tabela}")
    
    # Carregar dados da tabela
    df_tabela = pd.read_sql(f"SELECT * FROM {tabela}", conn)
    
    # Exibir dimensões
    print(f"\nDimensões da tabela {tabela}:", df_tabela.shape)
    
    # Exibir tipos das colunas
    print(f"\nTipos das colunas da tabela {tabela}:")
    display(df_tabela.dtypes)
    
    # Exibir as primeiras linhas
    print(f"\nPrimeiras linhas da tabela {tabela}:")
    display(df_tabela.head(5))
    
    # Estatísticas básicas
    print(f"\nEstatísticas descritivas da tabela {tabela}:")
    # Calcular as estatísticas descritivas
    estatisticas = df_tabela.describe(include="all")
    
    # Exibir as estatísticas formatadas
    display(estatisticas)
    print("\n")
    print("="*50)
    print("\n")

Analisando a tabela: Player

Dimensões da tabela Player: (11060, 7)

Tipos das colunas da tabela Player:


id                      int64
player_api_id           int64
player_name            object
player_fifa_api_id      int64
birthday               object
height                float64
weight                  int64
dtype: object


Primeiras linhas da tabela Player:


id  player_api_id         player_name  player_fifa_api_id  \
0   1         505942  Aaron Appindangoye              218353   
1   2         155782     Aaron Cresswell              189615   
2   3         162549         Aaron Doran              186170   
3   4          30572       Aaron Galindo              140161   
4   5          23780        Aaron Hughes               17725   

              birthday  height  weight  
0  1992-02-29 00:00:00  182.88     187  
1  1989-12-15 00:00:00  170.18     146  
2  1991-05-13 00:00:00  170.18     163  
3  1982-05-08 00:00:00  182.88     198  
4  1979-11-08 00:00:00  182.88     154


Estatísticas descritivas da tabela Player:


id  player_api_id player_name  player_fifa_api_id  \
count   11060.000000   11060.000000       11060        11060.000000   
unique           NaN            NaN       10848                 NaN   
top              NaN            NaN      Danilo                 NaN   
freq             NaN            NaN           7                 NaN   
mean     5537.511392  156582.427215         NaN       165664.910488   
std      3197.692647  160713.700624         NaN        58649.928360   
min         1.000000    2625.000000         NaN            2.000000   
25%      2767.750000   35555.500000         NaN       151889.500000   
50%      5536.500000   96619.500000         NaN       184671.000000   
75%      8306.250000  212470.500000         NaN       203883.250000   
max     11075.000000  750584.000000         NaN       234141.000000   

                   birthday        height        weight  
count                 11060  11060.000000  11060.000000  
unique                 5762           NaN           NaN  
top     1989-03-02 00:00:00           NaN           NaN  
freq                     10           NaN           NaN  
mean                    NaN    181.867445    168.380289  
std                     NaN      6.369201     14.990217  
min                     NaN    157.480000    117.000000  
25%                     NaN    177.800000    159.000000  
50%                     NaN    182.880000    168.000000  
75%                     NaN    185.420000    179.000000  
max                     NaN    208.280000    243.000000





Analisando a tabela: Match

Dimensões da tabela Match: (25979, 115)

Tipos das colunas da tabela Match:


id              int64
country_id      int64
league_id       int64
season         object
stage           int64
               ...   
GBD           float64
GBA           float64
BSH           float64
BSD           float64
BSA           float64
Length: 115, dtype: object


Primeiras linhas da tabela Match:


id  country_id  league_id     season  stage                 date  \
0   1           1          1  2008/2009      1  2008-08-17 00:00:00   
1   2           1          1  2008/2009      1  2008-08-16 00:00:00   
2   3           1          1  2008/2009      1  2008-08-16 00:00:00   
3   4           1          1  2008/2009      1  2008-08-17 00:00:00   
4   5           1          1  2008/2009      1  2008-08-16 00:00:00   

   match_api_id  home_team_api_id  away_team_api_id  home_team_goal  ...  \
0        492473              9987              9993               1  ...   
1        492474             10000              9994               0  ...   
2        492475              9984              8635               0  ...   
3        492476              9991              9998               5  ...   
4        492477              7947              9985               1  ...   

    SJA   VCH   VCD   VCA   GBH   GBD   GBA   BSH   BSD   BSA  
0  4.00  1.65  3.40  4.50  1.78  3.25  4.00  1.73  3.40  4.20  
1  3.80  2.00  3.25  3.25  1.85  3.25  3.75  1.91  3.25  3.60  
2  2.50  2.35  3.25  2.65  2.50  3.20  2.50  2.30  3.20  2.75  
3  7.50  1.45  3.75  6.50  1.50  3.75  5.50  1.44  3.75  6.50  
4  1.73  4.50  3.40  1.65  4.50  3.50  1.65  4.75  3.30  1.67  

[5 rows x 115 columns]


Estatísticas descritivas da tabela Match:


id    country_id     league_id     season         stage  \
count   25979.000000  25979.000000  25979.000000      25979  25979.000000   
unique           NaN           NaN           NaN          8           NaN   
top              NaN           NaN           NaN  2008/2009           NaN   
freq             NaN           NaN           NaN       3326           NaN   
mean    12990.000000  11738.630317  11738.630317        NaN     18.242773   
std      7499.635658   7553.936759   7553.936759        NaN     10.407354   
min         1.000000      1.000000      1.000000        NaN      1.000000   
25%      6495.500000   4769.000000   4769.000000        NaN      9.000000   
50%     12990.000000  10257.000000  10257.000000        NaN     18.000000   
75%     19484.500000  17642.000000  17642.000000        NaN     27.000000   
max     25979.000000  24558.000000  24558.000000        NaN     38.000000   

                       date  match_api_id  home_team_api_id  away_team_api_id  \
count                 25979  2.597900e+04      25979.000000      25979.000000   
unique                 1694           NaN               NaN               NaN   
top     2009-04-11 00:00:00           NaN               NaN               NaN   
freq                     60           NaN               NaN               NaN   
mean                    NaN  1.195429e+06       9984.371993       9984.475115   
std                     NaN  4.946279e+05      14087.453758      14087.445135   
min                     NaN  4.831290e+05       1601.000000       1601.000000   
25%                     NaN  7.684365e+05       8475.000000       8475.000000   
50%                     NaN  1.147511e+06       8697.000000       8697.000000   
75%                     NaN  1.709852e+06       9925.000000       9925.000000   
max                     NaN  2.216672e+06     274581.000000     274581.000000   

        home_team_goal  ...           SJA           VCH           VCD  \
count     25979.000000  ...  17097.000000  22568.000000  22568.000000   
unique             NaN  ...           NaN           NaN           NaN   
top                NaN  ...           NaN           NaN           NaN   
freq               NaN  ...           NaN           NaN           NaN   
mean          1.544594  ...      4.622343      2.668107      3.899048   
std           1.297158  ...      3.632164      1.928753      1.248221   
min           0.000000  ...      1.100000      1.030000      1.620000   
25%           1.000000  ...      2.500000      1.700000      3.300000   
50%           1.000000  ...      3.500000      2.150000      3.500000   
75%           2.000000  ...      5.250000      2.800000      4.000000   
max          10.000000  ...     41.000000     36.000000     26.000000   

                 VCA           GBH           GBD           GBA           BSH  \
count   22568.000000  14162.000000  14162.000000  14162.000000  14161.000000   
unique           NaN           NaN           NaN           NaN           NaN   
top              NaN           NaN           NaN           NaN           NaN   
freq             NaN           NaN           NaN           NaN           NaN   
mean        4.840281      2.498764      3.648189      4.353097      2.497894   
std         4.318338      1.489299      0.867440      3.010189      1.507793   
min         1.080000      1.050000      1.450000      1.120000      1.040000   
25%         2.550000      1.670000      3.200000      2.500000      1.670000   
50%         3.500000      2.100000      3.300000      3.400000      2.100000   
75%         5.400000      2.650000      3.750000      5.000000      2.620000   
max        67.000000     21.000000     11.000000     34.000000     17.000000   

                 BSD           BSA  
count   14161.000000  14161.000000  
unique           NaN           NaN  
top              NaN           NaN  
freq             NaN           NaN  
mean        3.660742      4.405663  
std         0.868272      3.189814  
min         1.330000   





Analisando a tabela: League

Dimensões da tabela League: (11, 3)

Tipos das colunas da tabela League:


id             int64
country_id     int64
name          object
dtype: object


Primeiras linhas da tabela League:


id  country_id                    name
0      1           1  Belgium Jupiler League
1   1729        1729  England Premier League
2   4769        4769          France Ligue 1
3   7809        7809   Germany 1. Bundesliga
4  10257       10257           Italy Serie A


Estatísticas descritivas da tabela League:


id    country_id                    name
count      11.000000     11.000000                      11
unique           NaN           NaN                      11
top              NaN           NaN  Belgium Jupiler League
freq             NaN           NaN                       1
mean    12452.090909  12452.090909                     NaN
std      8215.308472   8215.308472                     NaN
min         1.000000      1.000000                     NaN
25%      6289.000000   6289.000000                     NaN
50%     13274.000000  13274.000000                     NaN
75%     18668.000000  18668.000000                     NaN
max     24558.000000  24558.000000                     NaN





Analisando a tabela: Country

Dimensões da tabela Country: (11, 2)

Tipos das colunas da tabela Country:


id       int64
name    object
dtype: object


Primeiras linhas da tabela Country:


id     name
0      1  Belgium
1   1729  England
2   4769   France
3   7809  Germany
4  10257    Italy


Estatísticas descritivas da tabela Country:


id     name
count      11.000000       11
unique           NaN       11
top              NaN  Belgium
freq             NaN        1
mean    12452.090909      NaN
std      8215.308472      NaN
min         1.000000      NaN
25%      6289.000000      NaN
50%     13274.000000      NaN
75%     18668.000000      NaN
max     24558.000000      NaN





Analisando a tabela: Team

Dimensões da tabela Team: (299, 5)

Tipos das colunas da tabela Team:


id                    int64
team_api_id           int64
team_fifa_api_id    float64
team_long_name       object
team_short_name      object
dtype: object


Primeiras linhas da tabela Team:


id  team_api_id  team_fifa_api_id     team_long_name team_short_name
0   1         9987             673.0           KRC Genk             GEN
1   2         9993             675.0       Beerschot AC             BAC
2   3        10000           15005.0   SV Zulte-Waregem             ZUL
3   4         9994            2007.0   Sporting Lokeren             LOK
4   5         9984            1750.0  KSV Cercle Brugge             CEB


Estatísticas descritivas da tabela Team:


id    team_api_id  team_fifa_api_id team_long_name  \
count     299.000000     299.000000        288.000000            299   
unique           NaN            NaN               NaN            296   
top              NaN            NaN               NaN    Widzew Łódź   
freq             NaN            NaN               NaN              2   
mean    23735.301003   12340.521739      21534.305556            NaN   
std     15167.914719   25940.411135      42456.439408            NaN   
min         1.000000    1601.000000          1.000000            NaN   
25%      9552.500000    8349.000000        178.750000            NaN   
50%     22805.000000    8655.000000        673.500000            NaN   
75%     36250.500000    9886.500000       1910.750000            NaN   
max     51606.000000  274581.000000     112513.000000            NaN   

       team_short_name  
count              299  
unique             259  
top                GEN  
freq                 3  
mean               NaN  
std                NaN  
min                NaN  
25%                NaN  
50%                NaN  
75%                NaN  
max                NaN

## Tópico 4 - Entendimento e explicação dos dados, variáveis e contexto do problema.


## 4.1. Contexto do Dataset

O dataset utilizado neste projeto foi retirado do **Kaggle**, uma das maiores plataformas públicas de compartilhamento de datasets do mundo. Ele contém informações detalhadas sobre partidas, jogadores, times e ligas de futebol europeias, abrangendo um período de **10 anos**, de 2008 a 2018.

Nosso objetivo principal é **tentar prever o campeão da Premier League na temporada 2015/2016** utilizando estatísticas históricas. No entanto, um ponto importante do trabalho é mostrar que, apesar de termos acesso a grandes quantidades de dados, **prever resultados esportivos — especialmente um campeão — é algo extremamente difícil e muitas vezes próximo do impossível** devido à aleatoriedade do futebol.

Este projeto não busca apenas acertar o campeão, mas sim **demonstrar os limites da modelagem estatística em competições esportivas**.

---

## 4.2. Tabelas disponíveis e sua função

O dataset é composto por diversas tabelas que se relacionam entre si. Para este trabalho, utilizamos as seguintes:

### **1. Player — informações sobre jogadores**

Contém dados cadastrais e físicos de cada jogador, como nome, data de nascimento, altura e peso.

### **2. Match — informações sobre partidas**

É a **tabela mais importante do dataset** e o coração do projeto.
Ela conecta:

* jogadores
* times
* ligas
* países

Aqui estão resultados das partidas, escalações, estatísticas in-game, season, rodada, mandante, visitante etc.

### **3. League — informações sobre ligas**

Possui o nome e o país da liga.
Exemplo: Premier League (Inglaterra), La Liga (Espanha), Serie A (Itália), etc.

### **4. Country — países associados às ligas**

Tabela auxiliar que relaciona o país à liga e também fornece IDs únicos.

### **5. Team — informações dos times**

Contém nome completo, nome curto e identificadores utilizados pelo Kaggle e pela FIFA.

---

## 4.3. Variáveis principais (colunas) e seus significados

Aqui listamos as principais colunas que serão usadas na análise.
Abaixo estão os tipos das colunas e suas interpretações:

---

### 🔹 **Tabela Player**

```text
id                    int64  → ID interno
player_api_id         int64  → ID único do jogador no dataset
player_name           object → Nome do jogador
player_fifa_api_id    int64  → ID do jogador na base FIFA
birthday              object → Data de nascimento
height               float64 → Altura em centímetros
weight                int64  → Peso em libras
```

**Interpretação:**
Essa tabela serve como cadastro geral dos jogadores. As colunas numéricas permitem análises estatísticas, enquanto colunas como nome e aniversário são categóricas.

---

### 🔹 **Tabela Match** (115 colunas)

```text
id              int64  → ID da partida
country_id      int64  → País onde a liga pertence
league_id       int64  → Liga da partida
season          object → Temporada (ex: 2015/2016)
stage           int64  → Rodada (1 → abertura, 38 → última)
...
GBD           float64
GBA           float64
BSH           float64
BSD           float64
BSA           float64
```

**Interpretação:**
A tabela Match é extremamente rica. Além de mapeamento das partidas, ela contém:

* estatísticas avançadas sobre desempenho
* número de gols
* posse de bola
* finalizações
* escalações
* odds de apostas (colunas como BSH, BSD, BSA)

É a tabela que permite **construir séries históricas** e criar features para tentar prever o campeão.

---

### 🔹 **Tabela League**

```text
id             int64  → ID da liga
country_id     int64  → País associado
name          object → Nome da liga
```

**Interpretação:**
Tabela simples, mas essencial para filtrar apenas a **Premier League**, que é o foco do trabalho.

---

### 🔹 **Tabela Country**

```text
id       int64   → ID do país
name    object  → Nome do país
```

Serve como tabela auxiliar que permite cruzar ligas com países.

---

### 🔹 **Tabela Team**

```text
id                    int64    → ID da tabela interna
team_api_id           int64    → ID único do time
team_fifa_api_id    float64    → ID oficial (FIFA)
team_long_name       object    → Nome completo do time
team_short_name      object    → Nome curto do time
```

**Interpretação:**
Essencial para identificar os times na tabela Match e para análise de desempenho dos clubes ao longo da temporada.

---

## 4.4. Contexto do problema de Machine Learning

O problema que buscamos resolver é:

> **Com base em dados históricos de 2008 a 2015, conseguiríamos prever o campeão da Premier League 2015/2016?**

A resposta mais provável é **não** — e isso é justamente parte do aprendizado.
A temporada 2015/2016 foi histórica: o Leicester City, um time cotado para rebaixamento, foi o campeão. Essa é considerada a **maior zebra da história do futebol moderno**.

Nosso modelo tentará aprender:

* desempenho histórico dos times
* estatísticas das partidas
* tendências de resultados
* força do elenco

Mas mesmo com tudo isso, existe um limite do que o modelo consegue prever.

Tipo de problema:

* **Classificação multiclasse** (classe = campeão)
  ou
* **Regressão** (ex: prever pontuação final)

Decisão comum: transformar em classificação → prever qual time termina em 1º.

---

## 4.5. Relação entre dados e objetivo

A relação entre as tabelas e o problema é a seguinte:

* **Match** fornece resultados reais, estatísticas e pontos.
* **Team** permite identificar cada clube.
* **League** permite filtrar apenas a Premier League.
* **Player** permite enriquecer dados com informações de elenco.
* **Country** organiza a hierarquia liga → país.

Combinando essas tabelas, é possível:

* reconstruir a tabela de classificação de cada temporada
* analisar o desempenho de cada time
* criar features baseadas em gols, vitórias, derrotas, desempenho recente
* usar isso para tentar prever o campeão

No entanto, o futebol é caos puro — lesões, técnicos, motivação, sorte — tudo influencia.

O objetivo da análise é **mostrar tanto o poder quanto a limitação da ciência de dados em esportes imprevisíveis**.


## Tópico 5 - Visualização dos dados com gráficos apropriados (histogramas, boxplots, dispersão, etc.).


## Tópico 6 - Tratamento de dados inválidos ou inconsistentes.


## Tópico 7 - Tratamento de dados faltantes (remoção ou imputação).


## Tópico 8 - Seleção de atributos relevantes para o modelo.


## Tópico 9 - Classificação das variáveis (numéricas, categóricas, discretas, contínuas).


## Tópico 10 - Separação entre atributos preditores (X) e variável-alvo ( y ).


## Tópico 11 - Transformações necessárias nas colunas (normalização, codificação, etc.).


## Tópico 12 - Padronização dos dados para modelagem (ex: StandardScaler, MinMaxScaler).


## Tópico 13 - Divisão dos dados em conjunto de treino e teste.


## Tópico 14 - Treinamento de ao menos um modelo de machine learning supervisionado adequado ao problema.


## Tópico 15 - Avaliação do modelo (ex: acurácia, precisão, recall, F1-score, matriz de confusão).


## Tópico 16 - Documentação clara e detalhada do código no próprio notebook (com comentários e células explicativas).


## Tópico 17 - Criação de uma apresentação resumida em slides (Google Slides, PowerPoint, Canva etc.).


## Tópico 18 - Gravação de um vídeo (máx. 10 minutos) com ambos os integrantes explicando o projeto, o código e os resultados obtidos.
